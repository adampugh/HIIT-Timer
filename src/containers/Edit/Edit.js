import React, { Component } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";

import Navbar from "../../components/UI/navbarLoggedIn";
import EditBanner from "./editBanner";
import EditGrid from "./editGrid";
import Footer from "../../components/UI/footer";


export class Edit extends Component {
    state = {
        editModalOpen: false,
        deleteModalOpen: false
    }

    openEditModal = () => {
        this.setState({
            editModalOpen: true
        });
    }

    closeEditModal = () => {
        this.setState({
            editModalOpen: false
        });
    }

    openDeleteModal = () => {
        this.setState({
            deleteModalOpen: true
        });
    }

    closeDeleteModal = () => {
        this.setState({
            deleteModalOpen: false
        });
    }


    render() {
        let editPage = !this.props.workouts[this.props.location.state.workoutIndex] ? 
            <Redirect to="/workouts" /> : (
                <div>
                    <Navbar />
                    <EditBanner />
                    <EditGrid 
                        workout={this.props.workouts[this.props.location.state.workoutIndex].exercises}
                        workoutId={this.props.location.state.workoutId}
                        editModalIsOpen={this.state.editModalOpen}
                        handleOpenEditModal={this.openEditModal}
                        handleCloseEditModal={this.closeEditModal}
                        deleteModalIsOpen={this.state.deleteModalOpen}
                        handleOpenDeleteModal={this.openDeleteModal}
                        handleCloseDeleteModal={this.closeDeleteModal}
                    />
                    <Footer />
                </div>
            );

        return editPage;
    }
}

const mapStateToProps = state => {
    return {
        workouts: state.reducer.workouts
    }
}

export default connect(mapStateToProps)(Edit);