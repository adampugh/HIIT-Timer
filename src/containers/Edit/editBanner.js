import React from "react";

import dumbbellIcon from "../../assets/photos/dumbbell-icon.png";

const bannerText = () => (
    <div className="bannerText">
        <div className="bannerText__container">
            <h2>
                Add exercises to create a full HIIT workout. Click the + button below to add a new exercise. 
            </h2>
            <img className="bannerText__image" src={dumbbellIcon} alt="arm icon" />
        </div>
    </div>
);

export default bannerText;