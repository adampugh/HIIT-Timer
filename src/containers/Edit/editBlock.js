import React from "react";
var moment = require("moment");
var momentDurationFormatSetup = require("moment-duration-format");

const editBlock = (props) => {
    let customStyle = {
        background: props.color
    }

    return (
        <div className="editBlock" style={customStyle}>
            <h1>{props.name}</h1>
            <h1>{moment.duration(props.time, "seconds").format("mm:ss")}</h1>
        </div>
    )
}

export default editBlock;