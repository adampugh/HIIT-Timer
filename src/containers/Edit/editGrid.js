import React, { Component } from "react";

import EditBlock from "./editBlock";
import EditModal from "../../components/UI/editModal";
import DeleteExerciseModal from "../../components/UI/deleteExerciseModal";

class editGrid extends Component {
    state = {
        selectedExerciseIndex: null
    }

    handleDeleteModal = (exerciseIndex) => {
        this.setState({
            selectedExerciseIndex: exerciseIndex
        });
        this.props.handleOpenDeleteModal()
    }

    render() {
        return (
            <div>
                <EditModal 
                    isOpen={this.props.editModalIsOpen}
                    handleCloseModal={this.props.handleCloseEditModal}
                    workoutId={this.props.workoutId}
                />
                <DeleteExerciseModal
                    isOpen={this.props.deleteModalIsOpen}
                    handleCloseModal={this.props.handleCloseDeleteModal} 
                    modalTitle="Exercise"
                    exerciseIndex={this.state.selectedExerciseIndex}
                    workoutId={this.props.workoutId}
                />
                <div className="edit__wrapper">
                    <div className="container">
                        <div className="edit__grid">
                            {this.props.workout.map((exercise, index) => 
                                <div key={`${exercise.name}${index}`}>
                                    <EditBlock 
                                        name={exercise.name}
                                        time={exercise.time}
                                        color={exercise.color}
                                    />
                                    <div className="editBlock__delete">
                                        <h3 onClick={() => this.handleDeleteModal(index)}><i className="far fa-trash-alt"></i> Delete</h3>
                                    </div>
                                </div>
                            )}
                            <div onClick={this.props.handleOpenEditModal} className="edit__addBlock">
                                <h1>Add an Exercise</h1>
                                <i className="fas fa-plus-circle"></i>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default editGrid;