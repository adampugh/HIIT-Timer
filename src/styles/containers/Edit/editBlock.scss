.editBlock {
    margin: 0 auto;
    text-align: center;
    justify-self: center;
    align-self: center;
    border: 5px solid white;
    background: $purple-gradient;
    border-radius: 20px;
    color: white;
    height: 200px;
    padding: 40px;
    width: 200px;
    cursor: pointer;
    @include boxshadow;
    i, svg {
        font-size: 40px;
    }
}

.editBlock__delete {
    text-align: center;
    color: grey;
    margin: 0 auto 40px auto;
    cursor: pointer;
    width: 250px;
}

@media only screen and (min-width: $desktop-breakpoint) {
    .editBlock {
        h1 {
            margin-top: 0;
        }
        width: 80%;
        max-width: 800px;
        height: 100px;
    }
}
