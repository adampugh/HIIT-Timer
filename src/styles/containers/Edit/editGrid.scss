.edit__grid {
    display: grid;
    grid-template-columns: 1fr;
    padding-bottom: 100px;
    width: 100%;
    a {
        text-decoration: none;
    }
}

.edit__wrapper {
    background: $light-grey;
}

.edit__addBlock {
    text-align: center;
    justify-self: center;
    border: 5px dashed grey;
    border-radius: 20px;
    margin-bottom: 80px;
    padding: 40px;
    width: 200px;
    height: 200px;
    cursor: pointer;
    svg, i {
        font-size: 40px;
    }
    
}

@media only screen and (min-width: $desktop-breakpoint) {

    .edit__addBlock {
        width: 80%;
        max-width: 800px;
        height: 100px;
        h1 {
            margin-top: 0;
        }
        &:hover {
            @include boxshadow;
        }
    }
}