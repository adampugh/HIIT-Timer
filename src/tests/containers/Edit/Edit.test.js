import React from "react";
import { shallow } from "enzyme"; 

import { Edit } from "../../../containers/Edit/Edit";
import workoutState from "../../fixtures/workouts";

test("should render Edit correctly", () => {
    let locationObj = {
        state: {
            workoutIndex: 0
        }
    };
    const wrapper = shallow(<Edit workouts={workoutState.workouts} location={locationObj} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render redirect if location props are not passed in Edit", () => {
    let locationObj = {
        state: {
            workoutIndex: undefined
        }
    }
    const wrapper = shallow(<Edit workouts={workoutState.workouts} location={locationObj}/>);
    expect(wrapper).toMatchSnapshot();
});
