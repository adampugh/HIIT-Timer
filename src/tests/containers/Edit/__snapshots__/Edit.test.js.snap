// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Edit correctly 1`] = `
<div>
  <Connect(Component) />
  <bannerText />
  <editGrid
    deleteModalIsOpen={false}
    editModalIsOpen={false}
    handleCloseDeleteModal={[Function]}
    handleCloseEditModal={[Function]}
    handleOpenDeleteModal={[Function]}
    handleOpenEditModal={[Function]}
    workout={
      Array [
        Object {
          "color": "#40b3e1",
          "id": "L2453",
          "name": "squats",
          "time": 60,
        },
        Object {
          "color": "#40b3e1",
          "id": "L2435y",
          "name": "pressups",
          "time": 60,
        },
      ]
    }
  />
  <footer />
</div>
`;

exports[`should render redirect if location props are not passed in Edit 1`] = `
<Redirect
  push={false}
  to="/workouts"
/>
`;
